src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -423,231 +646,72 @@ public void Abort()
             // TODO
         }
 
-        public void StartReadingRequestBody()
+        private unsafe IISAwaitable WriteToIISAsync(ReadOnlyBuffer<byte> buffer)
```

Comment:
It's unfortunate this is duplicated between the websocket logic and here.

Reply:
I could try to abstract out the calls to native, however it may be over engineering it.